tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py

```diff
@@ -38,185 +43,235 @@
-                logger.info(f"Installing Node.js and npm using package: {node_package_url}")
-
-                # Use the install_package method to handle the installation.
-                install_result = host_manager.install_package(host, node_package_url, system='macos')
+    logging.error(
+        "Save the Wazuh indexer username and password into the Wazuh manager keystore"
+    )
+    save_indexer_credentials_into_keystore(host_manager)
 
-                # Logging the result of installation attempt.
-                logger.info(f"Node.js and npm installation result on {host}: {install_result}")
-            else:
-                logger.info("Node.js and npm are already installed.")
 
-@pytest.fixture(scope='module')
-def setup_vulnerability_tests(host_manager: HostManager) -> Generator:
-    """Setup the vulnerability tests environment
+def configure_vulnerability_detection_and_wait_until_feeds_are_updated(
+    host_manager: HostManager, vulnerability_detection_enabled: bool
+) -> None:
+    """Configure the environment with the default Vulnerability Detection configuration.
 
     Args:
-        host_manager (HostManager): An instance of the HostManager class.
+        host_manager (HostManager): Instance of the HostManager class.
+        vulnerability_detection_enabled (bool): True if the vulnerability detection is enabled, False otherwise
     """
-    logger.error("Init setup of environment")
+    logging.error(
+        f"Configuring the environment: Vulnerability Detection Enabled: {vulnerability_detection_enabled}"
+    )
+    current_dir = os.path.dirname(__file__)
+    configurations_dir = os.path.join(current_dir, "configurations")
+    configurations_paths = {
+        "manager": os.path.join(configurations_dir, "manager.yaml"),
+        "agent": os.path.join(configurations_dir, "agent.yaml"),
+    }
+
+    configure_environment(
+        host_manager,
+        load_vulnerability_detector_configurations(
+            host_manager,
+            configurations_paths=configurations_paths,
+            enable=vulnerability_detection_enabled,
+        ),
+    )
 
-    # Configure managers and agents
-    logger.error("Getting backup of current configurations")
-    hosts_configuration_backup = backup_configurations(host_manager)
+    logging.error("Restarting managers")
+    host_manager.control_environment("restart", ["manager"], parallel=True)
 
-    logger.error("Configuring environment")
-    configure_environment(host_manager, load_vulnerability_detector_configurations(host_manager))
-    logger.error("Save the Wazuh indexer username and password into the Wazuh manager keystore")
-    save_indexer_credentials_into_keystore(host_manager)
+    if vulnerability_detection_enabled:
+        logging.error("Wait until Vulnerability Detector has update all the feeds")
+        # Replace when https://github.com/wazuh/wazuh-qa/issues/5173 is complete
+        # wait_until_vd_is_updated(host_manager)
+        time.sleep(300)
```

Comment:
We should replace this and merge this PR after #5173

Reply:
Done 5f5cff583